code/modules/projectiles/gun_system.dm

```diff
@@ -702,13 +727,13 @@ and you're good to go.
 		return
 
 
-	play_fire_sound(gun_user)
+	play_fire_sound(loc)
 	muzzle_flash(firing_angle, loc)
 	simulate_recoil(dual_wield, gun_user)
 
 	//This is where the projectile leaves the barrel and deals with projectile code only.
 	//vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
-	projectile_to_fire.fire_at(target, gun_user, src, projectile_to_fire.ammo.max_range, projectile_to_fire.ammo.shell_speed, firing_angle, suppress_light = CHECK_BITFIELD(flags_gun_features, GUN_SILENCED))
+	projectile_to_fire.fire_at(target, loc, src, projectile_to_fire.ammo.max_range, projectile_to_fire.ammo.shell_speed, firing_angle, suppress_light = CHECK_BITFIELD(flags_gun_features, GUN_SILENCED))
```

Comment:
Are you sure this will not break things

Reply:
Im sure, the param here just gets the turf of gun_user so as to avoid hitting the turf the firer is on. Changing this to loc will not change anything in the case for hand fired guns and will make sure bullets dont hit the gun if its in sentry or deployed form.

Reply:
alright